roofit/histfactory/src/MakeModelAndMeasurementsFast.cxx

```diff
@@ -83,7 +83,9 @@ RooFit::OwningPtr<RooWorkspace>
 RooStats::HistFactory::MakeModelAndMeasurementFast(RooStats::HistFactory::Measurement &measurement,
                                                    HistoToWorkspaceFactoryFast::Configuration const &cfg)
 {
+  TDirectory::TContext dirContext;
   std::unique_ptr<TFile> outFile;
+  gDirectory = nullptr; // Disable global registration of histograms this file. HistFactory places them explicitly later.
```

Comment:
```suggestion
  TDirectory::TContext dirContext{nullptr} // Disable global registration of histograms this file. HistFactory places them explicitly later.
  std::unique_ptr<TFile> outFile;
```
Since the next line does not allocated a `TFile` and thus does not change `gDirectory`

Reply:
However line 138:
```
outFile = std::make_unique<TFile>(outputFileName.c_str(), "recreate");
```
should probably be:
```
outFile = std::make_unique<TFile>(outputFileName.c_str(), "RECREATE_WITHOUT_GLOBALREGISTRATION") // note: casing does not matter.
gDirectory = nullptr;  // Make sure the files does not become the current directory.  We do not need TContext here since we have one on the outer scope.
```
Side note the `_WITHOUT_GLOBALREGISTRATION` is probably the only thing strictly necessary for the goal of this PR (i.e. since it is not registered, it is not on the list of Cleanups and thus will not be traversed on delete of the histograms).

Reply:
Oh, I had no idea this option existed! That's really good to know!